stdlib/_locale.pyi

```diff
@@ -10,7 +10,7 @@ LC_NUMERIC: int
 LC_ALL: int
 CHAR_MAX: int
 
-def setlocale(category: int, locale: str | Iterable[str | None] | None = None) -> str: ...
+def setlocale(__category: int, __locale: str | Iterable[str | None] | None = None) -> str: ...
```

Comment:
This works fine with keyword args on my machine (on Python 3.8 and Python 3.13):

```pycon
>>> import locale
>>> locale.setlocale(0, locale='foo')
'foo'
>>> locale.setlocale(category=0, locale='foo')
'foo'
```

Reply:
I see CI is failing for this one as well :)

Reply:
That's `locale`, not `_locale`. `_locale.setlocale` really takes only positional args: https://github.com/python/cpython/blob/cc18b886a51672c59622837a2b8e83bf6be28c58/Modules/_localemodule.c#L91. But of course people should normally use the `locale` version, and stubtest is also complaining here. I'll look into how to fix this.

Reply:
Oh, right. Maybe we re-export `_locale.setlocale` in `locale.setlocale` currently, whereas at runtime they're actually two different functions?

Reply:
Indeed, we re-export it in the stub: https://github.com/python/typeshed/blob/7f9b3ea6c354273ff6ef78c15f274d6e29becb22/stdlib/locale.pyi#L11

But that doesn't reflect what happens at runtime, where `locale.setlocale()` is a pure-Python wrapper around `_locale.setlocale()`: https://github.com/python/cpython/blob/cc18b886a51672c59622837a2b8e83bf6be28c58/Lib/locale.py#L600-L615

Reply:
Fixed. Also noticed that the argument type for `_locale.setlocale` was wrong; unlike the higher-level version it doesn't accept `Iterable[str]`.